README.md

```diff
@@ -37,17 +36,78 @@ Reverse geocoding is the opposite: returning a list of places near a given point
 - Flexible software architecture
 - Easy to contribute software patches and features to
 - Easy to set up and configure your own instance
-- No external dependencies (such as postgres)
+- No external dependencies (such as PostgreSQL)
 - Reliable, configurable & fast import process
 - Work equally well for a small city and the entire planet
 
-### Developer Documentation & API Access
+### How does it work?
+
+Magic! Well, like any geocoder, Pelias essentially combines [full text search](https://en.wikipedia.org/wiki/Full_text_search)
+techniques with knowledge of geography to quickly search over many millions of records, each representing some sort of location on Earth.
+
+The Pelias architecture has three main components and several smaller pieces.
+
+#### Data importers
+
+The importers filter, normalize, and ingest geographic datasets into the Pelias database. Currently there are 4 officially supported importers: [OpenStreetMap](https://github.com/pelias/openstreetmap/), [OpenAddresses](https://github.com/pelias/openaddresses/), [Who's on First](https://github.com/pelias/whosonfirst/), and [Geonames](https://github.com/pelias/geonames/).
+
+We are always discussing [supporting additional datasets](https://github.com/pelias/pelias/issues/254). Pelias users can also write their own importers, for example to import proprietary data into your own instance of Pelias.
+
+#### Database
+The underlying datastore that does most of the query heavy-lifting and powers our search results. We use [Elasticsearch](https://www.elastic.co/). Currently version 2.4 is supported, with [plans to support 5.x soon](https://github.com/pelias/pelias/issues/461).
+
+We've built a tool called [pelias-schema](https://github.com/pelias/schema/) that sets up Elasticsearch indices properly for Pelias.
+
+#### Frontend services
+
+This is where the actual geocoding smarts live, and are the components that users interact with when performing geocoding queries. The services are:
```

Comment:
"smarts" is maybe a bit too colloquial for anyone without a west-coast US understanding of English. Perhaps `This is the actual geocoding process, and includes the components that users interact with when performing geocoding queries.` or `This is the actual geocoding process part. It also includes the components like web services that users interact with when performing geocoding queries.`